crates/modules/logger/src/lib.rs

```diff
@@ -19,17 +21,30 @@ async fn logger_task(
 ) -> Result<CleanExit, ModuleError> {
     let mut receiver = ctx.get_receiver();
     let mut rx_config = ctx.get_config();
-    let mut logger = Logger::from_config(rx_config.read()?);
+    let sender = ctx.get_sender();
+
+    let res_logger = Logger::from_config(rx_config.read()?);
+
+    if res_logger.is_err() {
+        sender
+            .raise_warning("Failed to connect to syslog".into())
+            .await;
+    }
+
+    let mut logger = res_logger.unwrap_or_else(|e| e);
 
     loop {
         tokio::select! {
             r = shutdown.recv() => return r,
             _ = rx_config.changed() => {
-                logger = Logger::from_config(rx_config.read()?);
+                logger = Logger::from_config(rx_config.read()?).unwrap_or_else(|e| e);
```

Comment:
`from_config` returns `Result`, so I thing just:

```suggestion
                logger = Logger::from_config(rx_config.read()?)?;
```

should work?

Reply:
from_config returns Result<Self, Self>, .unwrap(|e| e)  doesn't early return. Err(Self) is used to add a warning to the module status describing failure to connect to syslog without causing the logger module to fail. It's a little weird, can go either direction with syslog failure causing module failure I don't know which fits pulsars design better.

Reply:
I agree it's a weird way of handling it but I don't have a better solution right now.

Probably you should also add the warning in the "changed-config" branch